core-bundle/src/Resources/contao/classes/DataContainer.php

```diff
@@ -1335,21 +1335,21 @@ public function invalidateCacheTags()
 
 	public function addPtableTags($strTable, $intId, &$tags)
 	{
-		if (empty($GLOBALS['TL_DCA'][$strTable]['config']['ptable']))
+		$ptable = $GLOBALS['TL_DCA'][$strTable]['list']['sorting']['mode'] == 5 ? $strTable : ($GLOBALS['TL_DCA'][$strTable]['config']['ptable'] ?? null);
+
+		if (!$ptable)
 		{
 			$tags[] = 'contao.db.' . $strTable;
 
 			return;
 		}
 
-		$ptable = $GLOBALS['TL_DCA'][$strTable]['config']['ptable'];
-
 		Controller::loadDataContainer($ptable);
 
 		$objPid = $this->Database->prepare('SELECT pid FROM ' . Database::quoteIdentifier($strTable) . ' WHERE id=?')
 								 ->execute($intId);
 
-		if (!$objPid->numRows)
+		if (!$objPid->numRows || $objPid->pid == '0')
```

Comment:
```suggestion
		if (!$objPid->numRows || $objPid->pid == 0)
```

Reply:
It's a string coming from the DB so...it's not consistent in the core anyway :)

Reply:
> int() fields should return an integer.

Only in PHP 8.1+, no (https://github.com/contao/contao/issues/4278#issuecomment-1067836255)?